DistroLauncher/OOBE.cpp

```diff
@@ -210,11 +211,45 @@ namespace DistributionInfo {
 				}
 
 				// We don't need to be hard real time precise.
-				Sleep(997);
+				Sleep(997); // NOLINT(readability-magic-numbers): only used here.
 			}
 			return false;
 		}
 
+		// Returns true if OOBE has to be launched in text mode.
+		// That might be the case due lack of graphics support in WSL
+		// or user requirement, by setting the environment variable
+		// LAUNCHER_FORCE_MODE, which can only be:
+		//	0 or unset or invalid = autodetection
+		//	1 = text mode
+		//	2 = GUI mode.
+		bool mustRunOOBEinTextMode() {
+			// has to consider the NULL-terminating.
+			const DWORD expectedSize = 2;
+			wchar_t value[expectedSize];
+			auto readResult = GetEnvironmentVariable(L"LAUNCHER_FORCE_MODE", value, expectedSize);
+			// var unset is not an error.
+			const bool unset = (readResult == 0 && GetLastError() == ERROR_ENVVAR_NOT_FOUND);
+			// more than one char + NULL, that's invalid. Like a string or a more than one digit number.
+			const bool notASingleChar = (readResult >= expectedSize || value[1] != NULL);
+			// Handle both in the same way: autodetect.
+			if (unset || notASingleChar) {
+				return !Helpers::WslGraphicsSupported();
+			}
+
+			// Expected result if the env var is correctly set:
+			// readResult == 1 && value[1] == NULL && value[0] in (0,1,2).
+			switch (value[0]) {
+			case L'1': // forced text mode.
+				return true;
+				break;
```

Comment:
nitpick: is it for the linter that you break here?

Reply:
Nope! That was just me forgetting that one. 😅 Sorry about the silly mistake.

Reply:
Fixed. I'm going to merge it now.